packages/gasket-cli/src/config/utils.js

```diff
@@ -104,21 +104,35 @@ function addDefaultPlugins(gasketConfig) {
  */
 async function addUserPlugins(gasketConfig) {
   try {
-    const pluginsDir = path.join(gasketConfig.root, 'plugins');
-    const files = await readdir(pluginsDir);
-    const moduleNames = files
-      .filter(fileName => jsExtension.test(fileName))
-      .map(fileName => {
-        const fileSansExtension = fileName.replace(jsExtension, '');
-        return path.join(pluginsDir, fileSansExtension);
-      });
+    const dirPathArray = [
+      path.join(gasketConfig.root, 'plugins'),
+      path.join(gasketConfig.root, 'src', 'plugins')
+    ];
+    let moduleNames = [];
+    for (let i = 0; i <= dirPathArray.length - 1; i++) {
+      try {
+        const files = await readdir(dirPathArray[i]);
+        const moduleNamesFilter = files
+          .filter(fileName => jsExtension.test(fileName))
+          .map(fileName => {
+            const fileSansExtension = fileName.replace(jsExtension, '');
+            return path.join(dirPathArray[i], fileSansExtension);
+          });
+        await moduleNames.push(moduleNamesFilter);
+      } catch (e) {
+        await moduleNames.push(gasketConfig.plugins.add || []);
+      }
+    }
 
+    moduleNames = [].concat(...moduleNames);
     const pluginsConfig = gasketConfig.plugins || {};
+    moduleNames = (pluginsConfig.add || []).concat(moduleNames);
+    moduleNames = moduleNames.filter((elem, pos) => moduleNames.indexOf(elem) === pos);
```

Comment:
What is this code for? Is it supposed to remove duplicates? Why would that occur?